master/buildbot/reporters/gerrit.py

```diff
@@ -401,42 +407,68 @@ def getProperty(build, name):
                     revision = None
 
             if project is not None and revision is not None:
-                self.sendCodeReview(project, revision, result)
+                self.sendCodeReview(project, revision, result, getProperty(build, "event.change.id"))
                 return
 
-    def sendCodeReview(self, project, revision, result):
-        gerrit_version = self.getCachedVersion()
-        if gerrit_version is None:
-            self.callWithVersion(
-                lambda: self.sendCodeReview(project, revision, result))
-            return
+    @defer.inlineCallbacks
+    def sendCodeReview(self, project, revision, result, change_id=None):
```

Comment:
I think it would just be simpler to create a subclass, and override the relevent method. In anycase, we need two method here rather than a big one splitted with a large if else block

Reply:
What should be subclasses? The idea was that `GerritStatusPush` now supports both http and ssh API.

Reply:
It would be more explicit to have a (unchanged) GerritStatusPush and a new GerritHttpStatusPush

- API wise, the only difference is whether there is an auth or not. It does not seem very intuitive.
- coding wise, the main method is a big if else. This would be much better in a overridden method.